### PR TITLE
alhommada week3 node.js

### PR DESCRIPTION
**Pull request by aalhommada**

### CODE REVIEW

File: week3/homework/src/index.js [hunk 1]
```diff
@@ -3,80 +3,81 @@
...
-    id: todos.length + 1,
-    name: req.body.name,
+  const todoItem = {
+    id: todo.length + 1,
+    name: req.body.description,
     done: false,
   };
-  todos.push(todo);
-  res.send(todos);
+  todo.push(todoItem);
```
Comment (gajduk) [inline week3/homework/src/index.js]: This is not saved. You have to explicitly write this to the file system.

File: week3/homework/src/index.js [hunk 2]
```diff
@@ -3,80 +3,81 @@
 const express = require('express');
 const Joi = require('joi');
 const app = express();
-const todos = require('./todos');
+const validateTodos = require('./validateTodos');
+const todo = require('./todo.json');
```
Comment (gajduk) [inline week3/homework/src/index.js]: Please dont read json files using require. It is a really bad practice. Use `fs.readFile` like we did in the lectures.

File: week3/homework/src/index.js (hunk 1 above)
Comment (aalhommada) [reply week3/homework/src/index.js]: Thanks i will update it

File: week3/homework/src/index.js (hunk 2 above)
Comment (aalhommada) [reply week3/homework/src/index.js]: Thanks . I will modify it

File: week3/homework/src/index.js [hunk 3]
```diff
@@ -1,84 +1,100 @@
...
+    } else {
+      const todo = JSON.parse(contents);
+      const todoItem = {
+        id: todo.length + 1,
+        description: req.body.description,
+        done: false,
+      };
+      todo.push(todoItem);
+      const newTodo = JSON.stringify(todo);
+      fs.writeFile('/todo.json', newTodo, error => {
```
Comment (gajduk) [inline week3/homework/src/index.js]: exactly like this 👍

Comment (aalhommada) [reply week3/homework/src/index.js]: Thanks a lot @gajduk ,  I learn alot during your review.

**Pull request closed without merging**